### PR TITLE
Aggressive Modernization

### PR DESCRIPTION
**Pull request by jherdman**
I'm looking into some deprecations in my application, and they look to be coming from this add-on. I found, however, I couldn't even build this add-on as some of its dependencies are so old.

This pull request modernizes a few things:

* Ember CLI
* Ember Sass
* Some testing approaches

### CODE REVIEW

File: app/components/chartist-chart.js [hunk 1]
```diff
@@ -40,40 +48,17 @@ export default Ember.Component.extend({
...
-    return Ember.String.capitalize(this.get('type'));
+  chartType: computed('type', function() {
+    return capitalize(this.get('type'));
   }),
 
   data: null,
-  options: UNDEF,
-  responsiveOptions: UNDEF,
+  options: null,
+  responsiveOptions: null,
```
Comment (jherdman) [inline app/components/chartist-chart.js]: JSHint _really_ hated your UNDEF constant.

File: app/components/chartist-chart.js [hunk 2]
```diff
@@ -96,6 +84,36 @@ export default Ember.Component.extend({
...
+    );
+
+    chart.on('created', this._createdEventHook);
+
+    this.set('chart', chart);
+
+    this._super(...arguments);
+  },
+
+  onData: observer('data', function() {
```
Comment (jherdman) [inline app/components/chartist-chart.js]: I'm fairly certain we can refactor away from this observer in a future pull request...

**Comment by jherdman** [comment]
I need to double check this against my production app, but things look good! Hopefully we can move forward with this.

**Comment by jherdman** [comment]
Works like a charm!

**Comment by tylergaw** [comment]
@jherdman This is awesome! Thank you so much. As you can tell, I've neglected this project. I took a look through the code and it looks killer. Do you want me to add you as a collaborator to this repo so you can just run with it? If not, I can get it merged up, tagged for release and published, etc.

**Comment by jherdman** [comment]
Yeah, that'd be great! I'll see what I can do to help out.

**Comment by tylergaw** [comment]
Cool, sent an invite.

This is 👍 from me. Do you want to bump the version in package.json before merging? My versioning was a little fishy on this. I feel like this PR could move it to `1.0.0` since this is out and folks are using it in production settings.

**Comment by jherdman** [comment]
Sure, that sounds reasonable. I'll push up a commit that updates the commit bits.

**Comment by jherdman** [comment]
Done

**Comment by jherdman** [comment]
@tylergaw I've cut the 1.0 tag. I don't believe I have permission to publish a release on NPM though. Would you mind taking care of that?

**Comment by tylergaw** [comment]
@jherdman sorry for the slowness,  finally pushed it to npm https://www.npmjs.com/package/ember-cli-chartist

**Pull request merged**